src/backup.rs

```diff
@@ -0,0 +1,149 @@
+use failure::Error;
+
+use adb_command::AdbCommand;
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct BackupOptions {
+    device_id: Option<String>,
+    applications: String,
+    shared_storage: String,
+    system_apps: String,
+    only_specified_apps: String,
+}
+
+impl BackupOptions {
+    pub fn default() -> Self {
+        BackupOptions {
+            device_id: None,
+            applications: "-noapk".to_string(),
+            shared_storage: "-noshared".to_string(),
+            system_apps: "-nosystem".to_string(),
+            only_specified_apps: "-all".to_string(),
+        }
+    }
+
+    pub fn with_device_id(self, device_id: String) -> Self {
+        BackupOptions {
+            device_id: Some(device_id),
+            ..self
+        }
+    }
+
+    pub fn with_applications(self) -> Self {
+        BackupOptions {
+            applications: "-apk".to_string(),
+            ..self
+        }
+    }
+
+    pub fn with_shared_storage(self) -> Self {
+        BackupOptions {
+            shared_storage: "-shared".to_string(),
+            ..self
+        }
+    }
+
+    pub fn with_system_apps(self) -> Self {
+        BackupOptions {
+            system_apps: "-system".to_string(),
+            ..self
+        }
+    }
+
+    pub fn with_only_specified_apps(self, apps: String) -> Self {
+        BackupOptions {
+            only_specified_apps: apps,
```

Comment:
how are the apps passed here? is it a simple string with seperated app names ("app1;app2;...)? if yes, would a vec or something not be better?

Reply:
Hmm tried always just one app and that works but can't figure out how to pass multiple apps. 
from [xda-developers](https://forum.xda-developers.com/galaxy-nexus/general/guide-phone-backup-unlock-root-t1420351)
> <packages...>
> Here you can list the package names (e.g. com.google.android.apps.plus) specifically that you would like to backup. Use this only if you're looking to backup a specific application.

and also on the dev-site of android is mentioned [package_names](https://developer.android.com/studio/command-line/adb.html) and because of that I thought, that you can provide a list of packe names but it looks like that you can just backup one specific app per time.

Reply:
we can leave it like it is right now, but in the future we could provide the functionality to specifiy multiple apps by just calling the command multiple times. I think this would be a good bonus target if we have time left or maybe for v2